src/test/unit_tests/test_schema_keywords.js

```diff
@@ -0,0 +1,104 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+const mocha = require('mocha');
+const { default: Ajv } = require('ajv');
+const schema_keywords = require('../../util/schema_keywords');
+const SensitiveString = require('../../util/sensitive_string');
+const mongodb = require('mongodb');
+const assert = require('assert');
+
+/**
+ * @typedef {import('ajv').KeywordCxt} KeywordCxt
+ */
+
+const ajv = new Ajv({ verbose: true, allErrors: true });
+
+const test_schema_keywords = {
+    $id: 'test_schema_keywords',
+    methods: {
+        params: {
+            type: 'object',
+            additionalProperties: false,
+            properties: {
+                key1: {
+                    date: true,
+                },
+                key2: {
+                    idate: true,
+                },
+                key3: {
+                    objectid: true,
+                },
+                key4: {
+                    binary: true,
+                },
+                key5: {
+                    wrapper: SensitiveString,
+                },
+            },
+        },
+    },
+};
+
+
+mocha.describe('Test Schema Keywords', function() {
+
+    mocha.before('Adding Schema And Keywords', async function() {
+        ajv.addSchema(test_schema_keywords);
+        ajv.addKeyword(schema_keywords.KEYWORDS.methods);
+        ajv.addKeyword(schema_keywords.KEYWORDS.date);
+        ajv.addKeyword(schema_keywords.KEYWORDS.idate);
+        ajv.addKeyword(schema_keywords.KEYWORDS.objectid);
+        ajv.addKeyword(schema_keywords.KEYWORDS.binary);
+        ajv.addKeyword(schema_keywords.KEYWORDS.wrapper);
+    });
+
+    mocha.it('Test keyword date', async function() {
+        const validator = ajv.getSchema('test_schema_keywords#/methods/params');
+        const should_pass = { key1: new Date() };
+        assert.strictEqual(validator(should_pass), true);
+        const should_fail = { key1: 'not_a_date' };
+        assert.notStrictEqual(validator(should_fail), true);
```

Comment:
Using notStringEqual on booleans seems a bit cumbersome, I would write 
```suggestion
        assert.strictEqual(validator(should_fail), false);
```